libs/gltfio/include/gltfio/Animator.h

```diff
@@ -106,13 +106,17 @@ class UTILS_PUBLIC Animator {
 
     // If "instance" is null, then this is the primary animator.
     Animator(FFilamentAsset const* asset, FFilamentInstance* instance);
-    ~Animator();
 
     Animator(const Animator& animator) = delete;
     Animator(Animator&& animator) = delete;
     Animator& operator=(const Animator&) = delete;
 
     AnimatorImpl* mImpl;
+public:
+    Animator(FilamentAsset *asset, FilamentInstance *instance) : Animator(reinterpret_cast<FFilamentAsset*>(asset), reinterpret_cast<FFilamentInstance*>(instance))
+    {
+    }
+    ~Animator();
```

Comment:
move this to the existing `public` section and do not inline the implementation. You also need to document what this ctor does, how to use it and explain that in this case the caller must `delete` it. You also need to explain the expected lifetime of the `asset` and the `instance` with respect to the lifetime of the Animator.

Because this constructor will be confusing you *must* explain in the documentation that the normal way of getting an Animator is to call `FilamentInstance::getAnimator()` and that this new ctor is for a specific use case.

Documentation is everything here.